### PR TITLE
Add back "[exporter/awsemfexporter]Split EMF log with larger than 100 buckets."

### PR DESCRIPTION
**Pull request by zzhlogin**
#### Description
This PR fix the flaky unit test in previous PR: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36336, and add back the implementation of splitting the emf log logic. 

#### Link to tracking issue
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/36727

#### Testing
Unit test updated and passed with 10 count:
```
go test -run TestAddToGroupedMetric -count 10 -tags=always

PASS
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/exporter/awsemfexporter	0.016s

```

### CODE REVIEW

**Review by mxiamxia — APPROVED** [review]
Thanks!

File: exporter/awsemfexporter/grouped_metric_test.go [hunk 1]
```diff
@@ -509,7 +550,11 @@ func TestTranslateUnit(t *testing.T) {
 	assert.Equal(t, "Count", v)
 }
 
-func generateTestMetricMetadata(namespace string, timestamp int64, logGroup, logStreamName, instrumentationScopeName string, metricType pmetric.MetricType) cWMetricMetadata {
+func generateTestMetricMetadata(namespace string, timestamp int64, logGroup, logStreamName, instrumentationScopeName string, metricType pmetric.MetricType, batchIndex ...int) cWMetricMetadata {
```
Comment (andrzej-stencel) [inline exporter/awsemfexporter/grouped_metric_test.go:553]: I'm not a big fan of simulating optional parameters with variadic functions. I think it makes it harder to understand the meaning of the code. It encourages questions like "can I pass more than one parameter there? what happens if I do?". It makes it impossible for the compiler to report an issue when the function is mistakenly called with more than one "optional" parameter. Why not refactor the function to accept a regular `batchIndex` parameter? 

This is an internal function, so it's not a big problem.

File: .chloggen/split-emf-log-when-buckets-larger-than-100.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: awsemfexporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Split EMF log to multiple log splits when buckets larger than 100.
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [242]
```
Comment (andrzej-stencel) [inline .chloggen/split-emf-log-when-buckets-larger-than-100.yaml:13]: This points to https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/242. I suppose you had something else in mind?

File: exporter/awsemfexporter/grouped_metric_test.go (hunk 1 above)
Comment (zzhlogin) [reply exporter/awsemfexporter/grouped_metric_test.go]: Thank you for the feedback! I’ve updated the function to use a regular batchIndex int parameter.

File: .chloggen/split-emf-log-when-buckets-larger-than-100.yaml (hunk 2 above)
Comment (zzhlogin) [reply .chloggen/split-emf-log-when-buckets-larger-than-100.yaml]: Sorry, I was meant to use PR number. Updated.

**Review by Aneurysm9 — APPROVED** [review]

**Review by andrzej-stencel — APPROVED** [review]

**Pull request merged**